### DISCUSSION TITLE
Symfony 6

### DISCUSSION THREAD

**Discussion by joem236** [GeniusesOfSymfony/WebSocketBundle#462] [Q&A] [answered]
Hello,

I am adding web sockets to our Symfony 6 application.  Is there going to be support for running in this version?

I have added the 3.14 version though it fails to pass on messages.  Granted there seem to be a number of deprecation warnings.

Thanks,

Joe

**Comment by mbabker** [comment, marked as answer]
3.14 included Symfony 6 support.

**Reply by joem236** [reply]
I am running the following:
Symfony 6.0.x
PHP 8.1.2

when the client connects I get the below error message:

[Symfony\Component\ErrorHandler\Error\FatalError]                                                                                                            
  Compile Error: Declaration of Ratchet\Session\Storage\VirtualSessionStorage::start() must be compatible with Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage::start(): bool 

After patching the ratchet library for this and the other places where this error appears I see that the client connects to the websocket.  The onSubscribe() function in my Topic does not seem to have been executed as I have entries to log the function and steps within it.

The client is disconnected right after the client sends a message.

I do see the following deprecation messages in the console though no errors:

13:34:19 INFO      [php] User Deprecated: Method "IteratorAggregate::getIterator()" might add "\Traversable" as a native return type declaration in the future. Do the same in implementation "Ratchet\RFC6455\Messaging\Message" now to avoid errors or add an explicit @return annotation to suppress this message.
[
  "exception" => ErrorException {
    #message: "User Deprecated: Method "IteratorAggregate::getIterator()" might add "\Traversable" as a native return type declaration in the future. Do the same in implementation "Ratchet\RFC6455\Messaging\Message" now to avoid errors or add an explicit @return annotation to suppress this message."
    #code: 0
    #file: "./vendor/symfony/error-handler/DebugClassLoader.php"
    #line: 328
    #severity: E_USER_DEPRECATED
    trace: {
      ./vendor/symfony/error-handler/DebugClassLoader.php:328 { …}
      ./vendor/ratchet/rfc6455/src/Messaging/MessageBuffer.php:206 { …}
      ./vendor/ratchet/rfc6455/src/Messaging/MessageBuffer.php:194 { …}
      ./vendor/cboden/ratchet/src/Ratchet/WebSocket/WsServer.php:153 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Session/SessionProvider.php:106 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Http/HttpServer.php:54 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php:112 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php:95 { …}
      ./vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php:123 { …}
      ./vendor/react/stream/src/Util.php:71 { …}
      ./vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php:123 { …}
      ./vendor/react/stream/src/DuplexResourceStream.php:196 { …}
      ./vendor/react/event-loop/src/StreamSelectLoop.php:245 { …}
      ./vendor/react/event-loop/src/StreamSelectLoop.php:212 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php:74 { …}
      ./vendor/gos/web-socket-bundle/src/Server/Type/WebSocketServer.php:75 { …}
      ./vendor/gos/web-socket-bundle/src/Server/ServerLauncher.php:42 { …}
      ./vendor/gos/web-socket-bundle/src/Command/WebsocketServerCommand.php:73 { …}
      ./vendor/symfony/console/Command/Command.php:291 { …}
      ./vendor/symfony/console/Application.php:1007 { …}
      ./vendor/symfony/framework-bundle/Console/Application.php:94 { …}
      ./vendor/symfony/console/Application.php:299 { …}
      ./vendor/symfony/framework-bundle/Console/Application.php:80 { …}
      ./vendor/symfony/console/Application.php:171 { …}
      ./vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:54 { …}
      ./vendor/autoload_runtime.php:29 { …}
      ./bin/console:11 {
        › 
        › require_once dirname(__DIR__).'/vendor/autoload_runtime.php';
        › 
      }
    }
  }
]
13:34:19 INFO      [php] User Deprecated: Method "Countable::count()" might add "int" as a native return type declaration in the future. Do the same in implementation "Ratchet\RFC6455\Messaging\Message" now to avoid errors or add an explicit @return annotation to suppress this message.
[
  "exception" => ErrorException {
    #message: "User Deprecated: Method "Countable::count()" might add "int" as a native return type declaration in the future. Do the same in implementation "Ratchet\RFC6455\Messaging\Message" now to avoid errors or add an explicit @return annotation to suppress this message."
    #code: 0
    #file: "./vendor/symfony/error-handler/DebugClassLoader.php"
    #line: 328
    #severity: E_USER_DEPRECATED
    trace: {
      ./vendor/symfony/error-handler/DebugClassLoader.php:328 { …}
      ./vendor/ratchet/rfc6455/src/Messaging/MessageBuffer.php:206 { …}
      ./vendor/ratchet/rfc6455/src/Messaging/MessageBuffer.php:194 { …}
      ./vendor/cboden/ratchet/src/Ratchet/WebSocket/WsServer.php:153 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Session/SessionProvider.php:106 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Http/HttpServer.php:54 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php:112 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php:95 { …}
      ./vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php:123 { …}
      ./vendor/react/stream/src/Util.php:71 { …}
      ./vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php:123 { …}
      ./vendor/react/stream/src/DuplexResourceStream.php:196 { …}
      ./vendor/react/event-loop/src/StreamSelectLoop.php:245 { …}
      ./vendor/react/event-loop/src/StreamSelectLoop.php:212 { …}
      ./vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php:74 { …}
      ./vendor/gos/web-socket-bundle/src/Server/Type/WebSocketServer.php:75 { …}
      ./vendor/gos/web-socket-bundle/src/Server/ServerLauncher.php:42 { …}
      ./vendor/gos/web-socket-bundle/src/Command/WebsocketServerCommand.php:73 { …}
      ./vendor/symfony/console/Command/Command.php:291 { …}
      ./vendor/symfony/console/Application.php:1007 { …}
      ./vendor/symfony/framework-bundle/Console/Application.php:94 { …}
      ./vendor/symfony/console/Application.php:299 { …}
      ./vendor/symfony/framework-bundle/Console/Application.php:80 { …}
      ./vendor/symfony/console/Application.php:171 { …}
      ./vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:54 { …}
      ./vendor/autoload_runtime.php:29 { …}
      ./bin/console:11 {
        › 
        › require_once dirname(__DIR__).'/vendor/autoload_runtime.php';
        › 
      }
    }
  }
]
13:34:19 INFO      [websocket]  disconnected
[
  "connection_id" => 770,
  "session_id" => "1341365478620baba4d0e02502066915",
  "storage_id" => "770",
  "user_identifier" => ""
]

**Reply by mbabker** [reply]
You have the latest versions of all the Ratchet libraries too, right?  IIRC all of the Symfony 6 and PHP 8.1 notices should’ve been handled with those.  Even so, the deprecations can be ignored until PHP 9 becomes a thing.

With the immediate disconnect, check for issues on the JavaScript side of the application.  Something might be hitting an error and closing the connection there.

For not hitting the topic class, you’d have to step through all the classes in the websocket chain to figure out what’s causing it to not be reached.  The classes in this bundle have a bunch of logging, so it could be something in the Ratchet stack is aborting early (possibly sending a close response, which would trigger the disconnect log message) if there’s nothing from the bundle being logged.

**Reply by joem236** [reply]
I am currently stepping through the websocket chain...should have something soon
The versions I'm running right now are:
gos/pubsub-router-bundle             v2.7.1             Symfony PubSub Router Bundle
gos/web-socket-bundle                v3.14.0            Symfony Web Socket Bundle
gos/websocket-client                 v1.5.0             WAMP client in PHP
cboden/ratchet                       v0.4.4             PHP WebSocket library
ratchet/rfc6455                      v0.3.1             RFC6455 WebSocket protocol handler

The same client works as expected when connection to a Ratchet only implementation on the same codebase.

current configuration files:
// config/packages/gos_web_socket.yaml
```
gos_web_socket:
  server:
    host: 0.0.0.0

    port: 8888

    router:
      resources:
        - resource: '%kernel.project_dir%/config/pubsub/websocket/*'
          type: 'glob'

```
// config/pubsub/websocket/routing.yaml
```
my_rpc:
  pattern: rpc/{method}
  callback: 'my.rpc'
  requirements:
    method: "[a-z_]+"

my_topic:
  pattern: vc/channel
  callback: 'my.secured.topic'


```
// fragment from config/services.yaml
```
App\WebSocket\Topic\MyTopic:
        tags:
            - { name: gos_web_socket.topic }
```

**Reply by mbabker** [reply]
What you have here seems fine.

I haven't updated https://github.com/GeniusesOfSymfony/WebsocketAppDemo in a while but nothing here looks too far different from how that's set up.

**Reply by joem236** [reply]
Thanks for your patience...there was something off with the way the JavaScript was implemented on our end that was causing the issue.
